extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergBaseLayout.java

```diff
@@ -79,8 +85,16 @@ protected IcebergTableLocationKey locationKey(
                     }
                 }
 
-                // Add the data instructions.
-                instructions.dataInstructions().ifPresent(builder::setSpecialInstructions);
+                // Add the data instructions if provided as part of the IcebergInstructions.
+                if (instructions.dataInstructions().isPresent()) {
+                    builder.setSpecialInstructions(instructions.dataInstructions().get());
+                } else {
+                    // Attempt to create data instructions from the properties collection and URI.
+                    final Object dataInstructions = dataInstructionsProvider.fromServiceLoader(fileUri);
```

Comment:
I wonder if it's best to create a provider per instance of properties map, as this code is doing, or just to pass in the properties for each access. Right now, the Loader is instantiated per catalog, and hence we're going to the service loader per catalog. I guess we don't expect users to make many catalogs, so it may be fine.